src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsalertsapi/resource/CreateAlertCodeIntTest.kt

```diff
@@ -0,0 +1,374 @@
+      .bodyValue(createAlertCodeRequest())
+      .headers(setAuthorisation(roles = listOf(ROLE_ALERTS_WRITER)))
+      .headers(setAlertRequestContext(username = USER_THROW_EXCEPTION))
+      .exchange()
+      .expectStatus().isEqualTo(HttpStatus.BAD_GATEWAY)
+      .expectBody(ErrorResponse::class.java)
+      .returnResult().responseBody
+
+    with(response!!) {
+      assertThat(status).isEqualTo(502)
+      assertThat(errorCode).isNull()
+      assertThat(userMessage)
+        .isEqualTo("Downstream service exception: Get user details request failed")
+      assertThat(developerMessage).isEqualTo("Get user details request failed")
+      assertThat(moreInfo).isNull()
+    }
+  }
+
+  @Test
+  fun `should populate created by using user_name claim`() {
+    val request = createAlertCodeRequest()
+
+    val alert = webTestClient.post()
+      .uri("/alert-codes")
+      .bodyValue(request)
+      .headers(setAuthorisation(user = TEST_USER, roles = listOf(ROLE_ALERTS_ADMIN), isUserToken = true))
+      .exchange()
+      .expectStatus().isCreated
+      .expectHeader().contentType(MediaType.APPLICATION_JSON)
+      .expectBody(AlertCode::class.java)
+      .returnResult().responseBody!!
+
+    with(alert) {
+      assertThat(createdBy)
+        .isEqualTo(TEST_USER)
+    }
+  }
+
+  @Test
+  fun `should populate created by using username claim`() {
+    val request = createAlertCodeRequest()
+
+    val alert = webTestClient.post()
+      .uri("/alert-codes")
+      .bodyValue(request)
+      .headers(setAuthorisation(user = TEST_USER, roles = listOf(ROLE_ALERTS_ADMIN), isUserToken = false))
+      .exchange()
+      .expectStatus().isCreated
+      .expectHeader().contentType(MediaType.APPLICATION_JSON)
+      .expectBody(AlertCode::class.java)
+      .returnResult().responseBody!!
+
+    with(alert) {
+      assertThat(createdBy)
+        .isEqualTo(TEST_USER)
+    }
+  }
+
+  @Test
+  fun `should populate created by using Username header`() {
+    val request = createAlertCodeRequest()
+
+    val alert = webTestClient.post()
+      .uri("/alert-codes")
+      .bodyValue(request)
+      .headers(setAuthorisation(roles = listOf(ROLE_ALERTS_ADMIN)))
+      .headers(setAlertRequestContext())
+      .exchange()
+      .expectStatus().isCreated
+      .expectHeader().contentType(MediaType.APPLICATION_JSON)
+      .expectBody(AlertCode::class.java)
+      .returnResult().responseBody!!
+
+    with(alert) {
+      assertThat(createdBy)
+        .isEqualTo(TEST_USER)
+    }
+  }
+
+  @Test
+  fun `should create new alert code`() {
+    val request = createAlertCodeRequest()
+    val alertCode = webTestClient.createAlertCode(request = request)
+    assertThat(alertCode).isNotNull
+    assertThat(alertCode.code).isEqualTo("CO")
+    assertThat(alertCode.description).isEqualTo("Description")
+    assertThat(alertCode.createdAt).isCloseToUtcNow(Assertions.within(3, ChronoUnit.SECONDS))
+  }
+
+  @Test
+  fun `409 conflict - active alert code exists`() {
```

Comment:
Really minor here. It's checking for any existing alert with the same code, not only active ones. Perhaps change the function name to `409 conflict -alert code exists`?

Reply:
Done.